petsc is failing:
https://gitlab.spack.io/spack/spack-packages/-/jobs/21839849

I see this:
```
/home/software/spack/[padded-to-256-chars]/linux-x86_64_v3/intel-oneapi-mpi-2021.18.0-nzpax66k6uidgp3hhqbpf7tlk7sfycu2/mpi/2021.18/bin/mpicc -c -fPIC -Wall -Wwrite-strings -Wno-unknown-pragmas -Wconversion -Wno-sign-conversion -Wno-float-conversion -Wno-implicit-float-conversion -Qunused-arguments -fstack-protector -fvisibility=hidden -O3  -I/tmp/root/spack-stage/spack-stage-petsc-3.25.1-er4esfmur42e6vadp3sfuehdl7xvvawv/spack-src/include -I/tmp/root/spack-stage/spack-stage-petsc-3.25.1-er4esfmur42e6vadp3sfuehdl7xvvawv/spack-src/arch-linux-c-opt/include    -MMD -MP /tmp/root/spack-stage/spack-stage-petsc-3.25.1-er4esfmur42e6vadp3sfuehdl7xvvawv/spack-src/arch-linux-c-opt/ftn/mat/impls/aij/seq/inodef.c -o arch-linux-c-opt/obj/ftn/mat/impls/aij/seq/inodef.o
/home/software/spack/[padded-to-256-chars]/linux-x86_64_v3/intel-oneapi-mpi-2021.18.0-nzpax66k6uidgp3hhqbpf7tlk7sfycu2/mpi/2021.18/bin/mpicc -c -fPIC -Wall -Wwrite-strings -Wno-unknown-pragmas -Wconversion -Wno-sign-conversion -Wno-float-conversion -Wno-implicit-float-conversion -Qunused-arguments -fstack-protector -fvisibility=hidden -O3  -I/tmp/root/spack-stage/spack-stage-petsc-3.25.1-er4esfmur42e6vadp3sfuehdl7xvvawv/spack-src/include -I/tmp/root/spack-stage/spack-stage-petsc-3.25.1-er4esfmur42e6vadp3sfuehdl7xvvawv/spack-src/arch-linux-c-opt/include    -MMD -MP /tmp/root/spack-stage/spack-stage-petsc-3.25.1-er4esfmur42e6vadp3sfuehdl7xvvawv/spack-src/arch-linux-c-opt/ftn/mat/impls/baij/mpi/baijmkl/mpibaijmklf.c -o arch-linux-c-opt/obj/ftn/mat/impls/baij/mpi/baijmkl/mpibaijmklf.o
/tmp/root/spack-stage/spack-stage-petsc-3.25.1-er4esfmur42e6vadp3sfuehdl7xvvawv/spack-src/arch-linux-c-opt/ftn/mat/impls/baij/mpi/baijmkl/mpibaijmklf.c:17:11: error: call to undeclared function 'MatCreateBAIJMKL'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
   17 |   *ierr = MatCreateBAIJMKL(MPI_Comm_f2c(*(comm)), *bs, *m, *n, *M_M, *M_N, *d_nz, d_nnz, *o_nz, o_nnz, A);
      |           ^
/tmp/root/spack-stage/spack-stage-petsc-3.25.1-er4esfmur42e6vadp3sfuehdl7xvvawv/spack-src/arch-linux-c-opt/ftn/mat/impls/baij/mpi/baijmkl/mpibaijmklf.c:17:11: note: did you mean 'MatCreateBAIJ'?
/tmp/root/spack-stage/spack-stage-petsc-3.25.1-er4esfmur42e6vadp3sfuehdl7xvvawv/spack-src/include/petscmat.h:447:29: note: 'MatCreateBAIJ' declared here
  447 | PETSC_EXTERN PetscErrorCode MatCreateBAIJ(MPI_Comm, PetscInt, PetscInt, PetscInt, PetscInt, PetscInt, PetscInt, const PetscInt[], PetscInt, const PetscInt[], Mat *);
      |                             ^
1 error generated.
make[3]: *** [gmakefile:205: arch-linux-c-opt/obj/ftn/mat/impls/baij/mpi/baijmkl/mpibaijmklf.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```


It looks like ```PETSC_HAVE_MKL_SPARSE``` is not being defined:
https://github.com/petsc/petsc/blob/7c659a15437d3cd30f8d3579281dbc105b160796/include/petscmat.h#L411

I attached the configure log: [configure.log](https://github.com/user-attachments/files/27362435/configure.log)

I will add a conflict in petsc for mkl 2026.0+

--- a/repos/spack_repo/builtin/packages/heffte/package.py
+++ b/repos/spack_repo/builtin/packages/heffte/package.py
@@ -67,8 +67,8 @@ class Heffte(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("fftw@3.3.8:", when="+fftw", type=("build", "run"))
     depends_on("intel-oneapi-mkl", when="+mkl", type=("build", "run"))
-    # mkl rename dfti.hpp to dft.hpp in 2026.0, which breaks heffte@2.4.1 and earlier
-    conflicts("^intel-oneapi-mkl@2026.0:", when="@:2.4.1+mkl")
+    # mkl renamed dfti.hpp to dft.hpp in 2026.0, which breaks heffte@2.4.1 and earlier
+    conflicts("^intel-oneapi-mkl@2026.0:", when="@:2.4.1")
     depends_on("cuda@8.0:", when="+cuda", type=("build", "run"))
     depends_on("hip@3.8.0:", when="+rocm", type=("build", "run"))
     depends_on("rocfft@3.8.0:", when="+rocm", type=("build", "run"))

--- a/repos/spack_repo/builtin/packages/heffte/package.py
+++ b/repos/spack_repo/builtin/packages/heffte/package.py
@@ -67,6 +67,8 @@ class Heffte(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("fftw@3.3.8:", when="+fftw", type=("build", "run"))
     depends_on("intel-oneapi-mkl", when="+mkl", type=("build", "run"))
+    # mkl rename dfti.hpp to dft.hpp in 2026.0, which breaks heffte@2.4.1 and earlier
+    conflicts("^intel-oneapi-mkl@2026.0:", when="@:2.4.1+mkl")
     depends_on("cuda@8.0:", when="+cuda", type=("build", "run"))
     depends_on("hip@3.8.0:", when="+rocm", type=("build", "run"))
     depends_on("rocfft@3.8.0:", when="+rocm", type=("build", "run"))

--- a/repos/spack_repo/builtin/packages/magma/package.py
+++ b/repos/spack_repo/builtin/packages/magma/package.py
@@ -94,6 +94,9 @@ class Magma(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("cuda_arch=75", when="@:2.5.0", msg="magma: cuda_arch=75 needs a version > 2.5.0")
     conflicts("cuda_arch=80", when="@:2.5.3", msg="magma: cuda_arch=80 needs a version > 2.5.3")
 
+    # MAGMA uses mkl_zcsrmv, which is not available in MKL 2026.0+ 
+    conflicts("^intel-oneapi-mkl@2026.0:", msg="magma: MKL 2026.0+ is not supported")
+
     patch("ibm-xl.patch", when="@2.2:2.5.0%xl")
     patch("ibm-xl.patch", when="@2.2:2.5.0%xl_r")
     patch("magma-2.3.0-gcc-4.8.patch", when="@2.3.0%gcc@:4.8")

--- a/repos/spack_repo/builtin/packages/magma/package.py
+++ b/repos/spack_repo/builtin/packages/magma/package.py
@@ -94,7 +94,7 @@ class Magma(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("cuda_arch=75", when="@:2.5.0", msg="magma: cuda_arch=75 needs a version > 2.5.0")
     conflicts("cuda_arch=80", when="@:2.5.3", msg="magma: cuda_arch=80 needs a version > 2.5.3")
 
-    # MAGMA uses mkl_zcsrmv, which is not available in MKL 2026.0+ 
+    # MAGMA uses mkl_zcsrmv, which is not available in MKL 2026.0+
     conflicts("^intel-oneapi-mkl@2026.0:", msg="magma: MKL 2026.0+ is not supported")
 
     patch("ibm-xl.patch", when="@2.2:2.5.0%xl")

--- a/repos/spack_repo/builtin/packages/petsc/package.py
+++ b/repos/spack_repo/builtin/packages/petsc/package.py
@@ -354,8 +354,6 @@ class Petsc(Package, CudaPackage, ROCmPackage):
     depends_on("mumps+mpi~int64+metis+parmetis+openmp", when="+mumps+metis+openmp")
     depends_on("scalapack", when="+mumps")
     depends_on("mkl", when="+mkl-pardiso")
-    # issue with detecting MKL sparse support in configuration
-    conflicts("^intel-oneapi-mkl@2026.0:", msg="PETSc does not support MKL 2026.0 and later")
     depends_on("fftw+mpi", when="+fftw+mpi")
     depends_on("suite-sparse", when="+suite-sparse")
     depends_on("libx11", when="+X")
@@ -653,6 +651,11 @@ class Petsc(Package, CudaPackage, ROCmPackage):
 
         if "+mkl-pardiso" in spec:
             options.append("--with-mkl_pardiso-dir=%s" % spec["mkl"].prefix)
+
+        # See https://github.com/spack/spack-packages/pull/4651
+        if spec.satisfies("^intel-oneapi-mkl@2026.0:"):
+            options.append("--with-mkl_sparse=0")
+            options.append("--with-mkl_sparse_optimize=0")
 
         # For the moment, HPDDM does not work as a dependency
         # using download instead

--- a/repos/spack_repo/builtin/packages/petsc/package.py
+++ b/repos/spack_repo/builtin/packages/petsc/package.py
@@ -354,6 +354,8 @@ class Petsc(Package, CudaPackage, ROCmPackage):
     depends_on("mumps+mpi~int64+metis+parmetis+openmp", when="+mumps+metis+openmp")
     depends_on("scalapack", when="+mumps")
     depends_on("mkl", when="+mkl-pardiso")
+    # issue with detecting MKL sparse support in configuration
+    conflicts("^intel-oneapi-mkl@2026.0:", msg="PETSc does not support MKL 2026.0 and later")
     depends_on("fftw+mpi", when="+fftw+mpi")
     depends_on("suite-sparse", when="+suite-sparse")
     depends_on("libx11", when="+X")

